src/main/java/dao/AuthorDaoImpl.java

```diff
@@ -0,0 +1,140 @@
+package dao;
+
+import dao.interfaces.AuthorDao;
+import dto.AuthorDto;
+import models.Author;
+import models.Book;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuthorDaoImpl implements AuthorDao {
+  private final Connection connection;
+
+  public AuthorDaoImpl(Connection connection) {
+    this.connection = connection;
+  }
+
+  /**
+   * Method for finding all authors
+   * @return all authors in database
+   */
+  @Override
+  public List<AuthorDto> findAll() {
+    String query = "Select * from author";
+
+    try(PreparedStatement preparedStatement = connection.prepareStatement(query)){
+      ResultSet resultSet = preparedStatement.executeQuery();
+
+      List<AuthorDto> authors = new ArrayList<>();
+
+      while (resultSet.next()){
+        String name = resultSet.getString(2);
+        String secondname = resultSet.getString(3);
+        String surname = resultSet.getString(4);
+
+        resultSet.getString(3);
+
+        authors.add(new AuthorDto(name, secondname, surname));
+      }
+
+      return authors;
+    } catch (SQLException e) {
+      return null;
+    }
+  }
+
+  /**
+   * The method of finding the middle age of the reader, according to some author
+   * @param nameOfAuthor name of author
+   * @return avarage age of user
+   */
+  @Override
+  public double getAvarageAgeOfUserByAuthor(String nameOfAuthor) {
+    String query = "SELECT AVG(age)\n" +
+            "FROM users join journal on users.id = book_id \n" +
+            "join copy_book on journal.book_id = copy_book.id " +
+            "join copy on copy_book.book_id = copy.book_id " +
+            "join author on copy.author_id = author.id " +
+            "where author.author_name = ?";
+    try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
```

Comment:
will this work if query returns 0 results? if not - need to handle empty result set case.